yarn-project/end-to-end/src/spartan/utils.ts

```diff
@@ -34,18 +40,28 @@ const directConfigSchema = z.object({
   K8S: z.literal('false'),
 });
 
-const envSchema = z.discriminatedUnion('K8S', [k8sConfigSchema, directConfigSchema]);
+const envSchema = z.discriminatedUnion('K8S', [k8sLocalConfigSchema, k8sGCloudConfigSchema, directConfigSchema]);
 
-export type K8sConfig = z.infer<typeof k8sConfigSchema>;
+export type K8sLocalConfig = z.infer<typeof k8sLocalConfigSchema>;
+export type K8sGCloudConfig = z.infer<typeof k8sGCloudConfigSchema>;
 export type DirectConfig = z.infer<typeof directConfigSchema>;
 export type EnvConfig = z.infer<typeof envSchema>;
 
-export function getConfig(env: unknown): EnvConfig {
-  return envSchema.parse(env);
+export function isK8sConfig(config: EnvConfig): config is K8sLocalConfig | K8sGCloudConfig {
+  return config.K8S === 'local' || config.K8S === 'gcloud';
 }
 
-export function isK8sConfig(config: EnvConfig): config is K8sConfig {
-  return config.K8S === 'true';
+export function isGCloudConfig(config: EnvConfig): config is K8sGCloudConfig {
+  return config.K8S === 'gcloud';
+}
+
+export function setupEnvironment(env: unknown): EnvConfig {
+  const config = envSchema.parse(env);
+  if (isGCloudConfig(config)) {
+    const command = `gcloud container clusters get-credentials ${config.CLUSTER_NAME} --region=${config.REGION}`;
```

Comment:
I was disappointed by this: I would've thought copying in the gcloud and kube configs would've sufficed; indeed, it does locally.

Reply:
Maybe it's a permissions thing, like `~/.ssh`?

Reply:
Could it be the file owner not being the same user id as in docker?

Reply:
Hm those are good ideas. The thing about it is that when I said it works locally, I meant I ran the test container locally as it does in the workflow and I didn't need to do this explicit `get-credentials`; it does seems that some permissions aren't getting setup/passed correctly when the test container runs from CI. The other oddity is that the kube config gets mounted in the same way and doesn't appear to have a problem.

But not worth futzing with now I'm afraid. Thanks to you both for having a look!